pkg/dao/repository_configs.go

```diff
@@ -239,7 +239,7 @@ func (r repositoryConfigDaoImpl) List(
 	order := convertSortByToSQL(pageData.SortBy, sortMap)
 
 	filteredDB.Order(order).Find(&repoConfigs).Count(&totalRepos)
-	filteredDB.Preload("Repository").Limit(pageData.Limit).Offset(pageData.Offset).Find(&repoConfigs)
+	filteredDB.Preload("Repository").Preload("LastSnapshot").Limit(pageData.Limit).Offset(pageData.Offset).Find(&repoConfigs)
```

Comment:
Could be worth adding a test that when calling List(), LastSnapshot is preloaded properly